fvm/env.go

```diff
@@ -532,6 +671,60 @@ func (e *hostEnv) ImplementationDebugLog(message string) error {
 	return nil
 }
 
+func (e *hostEnv) ProgramParsed(location common.Location, duration time.Duration) {
+	if e.ctx.Tracer != nil && e.transactionEnv != nil && e.transactionEnv.traceSpan != nil {
```

Comment:
`e.ctx.Tracer != nil && e.transactionEnv != nil && e.transactionEnv.traceSpan != nil` is repeated a few times. Could be a function.

Reply:
it is actually, just forgot to update these few methods, fixed now